### PR TITLE
refactor!: curried record/omit with immutable keys

### PR DESCRIPTION
**Pull request by baoshan**
This PR modifies the `record/omit` function:

1. Function is manually curried.
2. `keys` are immutable.
3. The non-empty requirement from `keys` is removed (please blame me if this is intentional).
4. The interface respects the TypeScript `Omit` utility type.

### CODE REVIEW

File: record.ts [hunk 1]
```diff
@@ -93,15 +93,31 @@ export function deleteAt<K extends string>(
...
+ * @example
+ * ```ts
+ * import { omit } from "./record.ts";
+ * omit(["a", "c"])({ a: 1, b: 2 }) // { b: 2 }
+ * ```
+ *
+ * @category combinators
+ */
+// deno-lint-ignore no-explicit-any
+export function omit<T, K extends keyof any>(
```
Comment (baetheus) [inline record.ts:110]: Shouldn't K extend keyof T?

**Review by baetheus — CHANGES_REQUESTED** [review]

File: record.ts (hunk 1 above)
Comment (baoshan) [reply record.ts:110]: I wish the interface respects the original definition of TypeScript’s `Omit` utility type, which doesn’t pose constraints on `K`.

```
/**
 * Construct a type with the properties of T except for those in type K.
 */
type Omit<T, K extends keyof any> = Pick<T, Exclude<keyof T, K>>;
```

**Review by baetheus — DISMISSED** [review]

**Comment by baoshan** [comment]
Is there anything that blocks this PR from landing? Please let me know if there’s something I can do.

**Review by pixeleet — CHANGES_REQUESTED** [review]
@baoshan formatting check is failing for some reason I just realised

**Comment by baoshan** [comment]
Chunks of files need to be re-formatted. So I append a new commit.

**Pull request merged**